Frontend/js/src/WebSocketClient.js

```diff
@@ -0,0 +1,115 @@
+/* global: window */
+
+import sessionService from './sessionService.js';
+import EventTarget from './EventTarget.js';
+
+const location = window.location;
+
+/**
+ * Encapsulate WebSocket and provides the endpoint, filtering stream and authentification status.
+ * Events:
+ * - open
+ * - error
+ * - close
+ * - authed
+ * - token
+ * - *command*
+ * @author Adam Wegrzynek <adam.wegrzynek@cern.ch>
+ * @author Vladimir Kosmala <vladimir.kosmala@cern.ch>
+ */
+export default class WebSocketClient extends EventTarget {
+  /**
+   * Create a connection to the server
+   */
+  constructor() {
+    super();
+
+    this.connection = null;
+    this.authed = false;
+
+    this._connect();
+  }
+
+  /**
+   * Private. Create an instance of WebSocket and binds events
+   */
+  _connect() {
+    const session = sessionService.get();
+    const url = new URL(location);
+    url.protocol = 'wss';
+    url.searchParams.append('token', session.token);
+
+    this.connection = new WebSocket(url);
+    this.connection.addEventListener('message', (...args) => this._handleProtocolMessages(...args));
+    this.connection.addEventListener('open', (...args) => this.dispatchEvent(...args));
+    this.connection.addEventListener('error', (...args) => this.dispatchEvent(...args));
+    this.connection.addEventListener('close', (...args) => this.dispatchEvent(...args));
+  }
+
+  /**
+   * Private. Handle non-user messages: authentification, token refresh, errors
+   * @param {event} e - Event message received by websocket
+   */
+  _handleProtocolMessages(e) {
+    let parsed;
+
+    try {
+      parsed = JSON.parse(e.data);
+    } catch (e) {
+      throw new Error(`unable to parse ws data`);
+    }
+
+    // handling authentification success
+    if (parsed.command == 'authed') {
+      this.authed = true;
+      this.dispatchEvent(new Event('authed', parsed));
+      return;
+    }
+
+    // handling token refresh error
```

Comment:
After our discussion I actually disabled token refresh but the code can stay in case